### PR TITLE
feat(github-invite): link to open modal

### PR DESCRIPTION
**Pull request by cathteng**
The modal should be shown when a user hits the members list with a query param in the url.

Also fixes some followups from #55003.

### CODE REVIEW

File: static/app/views/settings/organizationMembers/inviteBanner.tsx [hunk 1]
```diff
@@ -68,6 +70,20 @@ export function InviteBanner({
     });
   }, [api, organization, promptsFeature, hideBanner]);
 
+  useEffect(() => {
+    const params = new URLSearchParams(location.search);
+    const showModal = params.get('showModal');
```
Comment (cathteng) [inline static/app/views/settings/organizationMembers/inviteBanner.tsx:75]: i'm not a fan of the name. any suggestions?

Comment (evanpurkhiser) [inline static/app/views/settings/organizationMembers/inviteBanner.tsx:75]: I would probably just use the `query-string` package for this

File: static/app/views/settings/organizationMembers/inviteBanner.tsx [hunk 2]
```diff
@@ -305,9 +316,7 @@ const MemberCardContentRow = styled('div')`
   align-items: center;
   margin-bottom: ${space(0.25)};
   font-size: ${p => p.theme.fontSizeSmall};
-  & > *:first-child {
-    margin-right: ${space(0.75)};
-  }
+  gap: ${space(0.75)};
 `;
```
Comment (evanpurkhiser) [inline static/app/views/settings/organizationMembers/inviteBanner.tsx:320]: Should this have been `ButtonBar`?

File: static/app/views/settings/organizationMembers/inviteBanner.tsx [hunk 3]
```diff
@@ -68,6 +70,20 @@ export function InviteBanner({
...
+
+    if (!hideBanner && showModal) {
+      openInviteMissingMembersModal({
+        allowedRoles,
+        missingMembers,
+        organization,
+        onClose: onModalClose,
+      });
+    }
+  }, [allowedRoles, missingMembers, organization, onModalClose, location, hideBanner]);
```
Comment (evanpurkhiser) [inline static/app/views/settings/organizationMembers/inviteBanner.tsx:85]: That's a lot of stuff that might re-open the modal if it changes reference.

File: static/app/components/modals/inviteMissingMembersModal/index.tsx [hunk 4]
```diff
@@ -59,7 +59,7 @@ export function InviteMissingMembersModal({
 
   const api = useApi();
 
-  if (!memberInvites || !organization.access.includes('org:write')) {
+  if (!memberInvites.length || !organization.access.includes('org:write')) {
```
Comment (sentaur-athena) [inline static/app/components/modals/inviteMissingMembersModal/index.tsx:62]: js has weird ways of treating objects as boolean. Can you make this more clear like
`memberInvites && memberInvites.length > 0`?

File: static/app/views/settings/organizationMembers/inviteBanner.tsx (hunk 2 above)
Comment (cathteng) [reply static/app/views/settings/organizationMembers/inviteBanner.tsx:316]: i made the ButtonContainer a `ButtonBar`

File: static/app/views/settings/organizationMembers/inviteBanner.tsx (hunk 3 above)
Comment (sentaur-athena) [reply static/app/views/settings/organizationMembers/inviteBanner.tsx:86]: @cathteng is this actually expected? You're using this to pass params to openInviteMissingMembersModal but useEffect params are basically the ones you want to trigger useEffect on change, so only hideBanner. You can instead call a local with no params.

Comment (cathteng) [reply static/app/views/settings/organizationMembers/inviteBanner.tsx:86]: call a local function with no params? wouldn't i still have to use `useCallback` with dependencies on these same variables?

File: static/app/components/modals/inviteMissingMembersModal/index.tsx [hunk 5]
```diff
@@ -59,7 +58,10 @@ export function InviteMissingMembersModal({
 
   const api = useApi();
 
-  if (!memberInvites || !organization.access.includes('org:write')) {
+  if (
+    (memberInvites && memberInvites.length === 0) ||
```
Comment (davidenwang) [inline static/app/components/modals/inviteMissingMembersModal/index.tsx:62]: do you need the `memberInvites &&` part here? Seems like given the types, memberInvites will not be null/undefined.

File: static/app/views/settings/organizationMembers/inviteBanner.tsx [hunk 6]
```diff
@@ -157,19 +176,8 @@ export function InviteBanner({
...
-            onClick={() =>
-              openInviteMissingMembersModal({
-                allowedRoles,
-                missingMembers,
-                onClose: onModalClose,
-                organization,
-              })
-            }
-          >
+        <ButtonBar gap={1}>
```
Comment (davidenwang) [inline static/app/views/settings/organizationMembers/inviteBanner.tsx:179]: nice, button bar is good

File: static/app/views/settings/organizationMembers/inviteBanner.tsx [hunk 7]
```diff
@@ -68,6 +80,15 @@ export function InviteBanner({
     });
   }, [api, organization, promptsFeature, hideBanner]);
 
+  useEffect(() => {
+    const params = new URLSearchParams(location.search);
```
Comment (davidenwang) [inline static/app/views/settings/organizationMembers/inviteBanner.tsx:84]: i do think that using the `query-string` package here is more inline with the rest of the frontend codebase, and would just end up looking something like

```
const inviteMissingMembers = qs.parse(location.search).inviteMissingMembers
```

but I asked in #discuss-frontend if there is a super hard preference.

**Review by davidenwang — APPROVED** [review]

File: static/app/views/settings/organizationMembers/inviteBanner.tsx (hunk 7 above)
Comment (cathteng) [reply static/app/views/settings/organizationMembers/inviteBanner.tsx:84]: i can use qs

File: static/app/components/modals/inviteMissingMembersModal/index.tsx (hunk 5 above)
Comment (cathteng) [reply static/app/components/modals/inviteMissingMembersModal/index.tsx:62]: i think you're right

**Pull request merged**